meshtastic/interfaces/ble/lifecycle_ownership_runtime.py

```
@@ -0,0 +1,539 @@
+                do_transition_to_disconnected=do_transition_to_disconnected,
+            )
+        if should_publish_disconnect and not is_closing:
+            with iface._state_lock:
+                publish_disconnect = (
+                    getattr(iface, "_connection_session_epoch", 0)
+                    == disconnect_session_epoch
+                )
+            if publish_disconnect:
+                iface._disconnected()
+
+    def _verify_and_publish_connected(
```

Ruff (0.15.6)

[warning] Too many statements (61 > 50)

(PLR0915)